nixos/modules/services/misc/matrix-conduit.nix

```diff
@@ -0,0 +1,233 @@
+              type = types.ints.positive;
+              default = 20000000;
+              description = "Max request size in bytes. Don't forget to also change it in the proxy.";
+            };
+            global.allow_registration = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Whether new users can register on this server.";
+            };
+            global.allow_encryption = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Whether new encrypted rooms can be created. Note: existing rooms will continue to work.";
+            };
+            global.allow_federation = mkOption {
+              type = types.bool;
+              default = false;
+              description = ''
+                Whether this server federates with other servers.
+                Upstream documentation contains the following warning:
+                You should not join Matrix rooms without asking the admins first.
+                We do not know whether Conduit is safe for general use yet, so you
+                should assume there is some chance that it breaks rooms permanently
+                for all participating users. We are not aware of such a bug today,
+                but we would like to do more testing.
+              '';
+            };
+            global.allow_jaeger = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Whether support monitoring and troubleshooting through jaeger is enabled on this server.";
+            };
+            global.trusted_servers = mkOption {
+              type = types.listOf types.str;
+              default = [ "matrix.org" ];
+              description = "Servers trusted with signing server keys.";
+            };
+
+            global.address = mkOption {
+              type = types.str;
+              default = "::1";
+              description = "Address to listen on for connections by the reverse proxy/tls terminator.";
+            };
+            global.database_path = mkOption {
+              type = types.str;
+              default = "/var/lib/matrix-conduit/conduit.db";
+              readOnly = true;
+              description = ''
+                Path to the conduit database, the directory where conduit will save its data.
+                Note that due to using the DynamicUser feature of systemd, this value should not be changed
+                and is set to be read only.
+              '';
+            };
+          };
+        };
+        default = {};
+        description = ''
+            Generates the conduit.toml configuration file. Refer to
+            <link xlink:href="https://gitlab.com/famedly/conduit/-/blob/master/conduit-example.toml"/>
+            for details on supported values.
+            Note that database_path should not be set because the services reliance on systemd StateDir.
```

Comment:
```suggestion
            Note that database_path can not be set because the service's reliance on systemd StateDir.
```

Reply:
I don't think this is resolved.

Reply:
It was at some point. Must have made a mistake while rebasing.